.travis.yml

```diff
@@ -1,32 +1,69 @@
-# validate this file at http://lint.travis-ci.org/
-# ocaml and coq are not on the list of languages:
-language:       generic
-sudo: 		required
-# "trusty" is the most modern release of Ubuntu supported
-dist: 		trusty
-git:
-  submodules:   false
-install:
-  - sudo add-apt-repository -y ppa:avsm
-  - sudo apt-get update
-  - sudo apt-get install -y opam aspcud time libgtk2.0-dev libgtksourceview2.0-dev emacs
-# build coqide along with the Tactics package, just because it's a short package
+language: generic
+sudo: false
+
+services:
+  - docker
+
 env:
-  - PACKAGES="Foundations Combinatorics Algebra NumberSystems PAdics"
-  - PACKAGES=CategoryTheory
-  - PACKAGES="MoreFoundations Ktheory"
-  - PACKAGES=HomologicalAlgebra
-  - PACKAGES=Topology
-  - PACKAGES=RealNumbers
-  - PACKAGES=SubstitutionSystems
-  - PACKAGES=Tactics BUILD_COQIDE=yes BUILD_ALSO=TAGS
-  - PACKAGES=Folds
-  - FOUNDATIONS_CHANGE_ERROR=yes BUILD_ALSO="check-for-change-to-Foundations enforce-listing-of-proof-files"
-# building Coq in a separate stage folds up the output in the log:
+  global:
+    - THIS_REPO=UniMath
+  matrix:
+    - PACKAGES="Foundations Combinatorics Algebra NumberSystems PAdics"
+    - PACKAGES=CategoryTheory
+    - PACKAGES="MoreFoundations Ktheory"
+    - PACKAGES=HomologicalAlgebra
+    - PACKAGES=Topology
+    - PACKAGES=RealNumbers
+    - PACKAGES=SubstitutionSystems
+    - PACKAGES=Tactics BUILD_COQIDE=yes BUILD_ALSO=TAGS
+    - PACKAGES=Folds
+    - FOUNDATIONS_CHANGE_ERROR=yes BUILD_ALSO="check-for-change-to-Foundations enforce-listing-of-proof-files"
+
+# The "docker run" command will pull if needed.
+# Running this first gives two tries in case of network lossage.
 before_script:
-  - opam init --yes --no-setup
-  - eval $(opam config env)
-  - opam install lablgtk camlp5 num --yes --verbose
-  - time make build-coq
+  - timeout 5m docker pull palmskog/xenial-unimath || true
```

Comment:
Where does docker find "palmskog/xenial-unimath" ?

If the command fails, why ignore the failure?

Reply:
Oh, I guess it finds it as one of your public repositories at docker.com.  How can I get a web page that displays your public repositories?

Reply:
Regarding ignoring a failed `docker pull` command, note that the `docker run` command later in the script will also download the Docker image. In this way, there will be two attempts to download the image if network connectivity is bad.

Reply:
Okay, I've created a docker organization called UniMath, so we can have our own docker image.  See https://hub.docker.com/u/UniMath/.

@benediktahrens -- make a docker.com account and tell me the user name, and I'll add you as an owner.  (The github notions of user, organization, owner, and team seem the same on docker.)